src/Fl_Check_Browser.cxx

```diff
@@ -300,90 +299,89 @@ int Fl_Check_Browser::remove(int item) {
 
 /**  Remove every item from the browser.*/
 void Fl_Check_Browser::clear() {
-        cb_item *p = first;
-        cb_item *next;
-
-        if (p == 0) {
-                return;
-        }
-
-        new_list();
-        do {
-                next = p->next;
-                free(p->text);
-                free(p);
-                p = next;
-        } while (p);
-
-        first = last = 0;
-        nitems_ = nchecked_ = 0;
-        cached_item = -1;
+  cb_item *p = first;
+  cb_item *next;
+
+  if (!p) return;
+
+  new_list();
+  do {
+    next = p->next;
+    free(p->text);
+    free(p);
+    p = next;
+  } while (p);
+
+  first = last = 0;
+  nitems_ = nchecked_ = 0;
+  cached_item = -1;
 }
 
 /** Gets the current status of item item. */
 int Fl_Check_Browser::checked(int i) const {
-        cb_item *p = find_item(i);
+  cb_item *p = find_item(i);
 
-        if (p) return p->checked;
-        return 0;
+  if (p) return p->checked;
+  return 0;
 }
 
 /** Sets the check status of item item to b. */
 void Fl_Check_Browser::checked(int i, int b) {
-        cb_item *p = find_item(i);
-
-        if (p && (p->checked ^ b)) {
-                p->checked = b;
-                if (b) {
-                        nchecked_++;
-                } else {
-                        nchecked_--;
-                }
-                redraw();
-        }
+  cb_item *p = find_item(i);
+
+  if (p && (p->checked ^ b)) {
+    p->checked = b;
+    if (b) {
+      nchecked_++;
+    } else {
+      nchecked_--;
+    }
+    redraw();
+  }
 }
 
 /**  Returns the index of the currently selected item.*/
 int Fl_Check_Browser::value() const {
-        return lineno((cb_item *)selection());
+  return lineno((cb_item *)selection());
 }
 
 /**  Return a pointer to an internal buffer holding item item's text.*/
 char *Fl_Check_Browser::text(int i) const {
-        cb_item *p = find_item(i);
+  cb_item *p = find_item(i);
 
-        if (p) return p->text;
-        return 0;
+  if (p) return p->text;
+  return 0;
 }
 
 /**  Sets all the items checked.*/
 void Fl_Check_Browser::check_all() {
-        cb_item *p;
+  cb_item *p;
 
-        nchecked_ = nitems_;
-        for (p = first; p; p = p->next) {
-                p->checked = 1;
-        }
-        redraw();
+  nchecked_ = nitems_;
+  for (p = first; p; p = p->next) {
+    p->checked = 1;
+  }
+  redraw();
 }
 
 /**  Sets all the items unchecked.*/
 void Fl_Check_Browser::check_none() {
-        cb_item *p;
+  cb_item *p;
 
-        nchecked_ = 0;
-        for (p = first; p; p = p->next) {
-                p->checked = 0;
-        }
-        redraw();
+  nchecked_ = 0;
+  for (p = first; p; p = p->next) {
+    p->checked = 0;
+  }
+  redraw();
 }
 
 int Fl_Check_Browser::handle(int event) {
   if (event == FL_PUSH) {
     int X, Y, W, H;
     bbox(X, Y, W, H);
-    if (Fl::event_inside(X, Y, W, H))
+    if (Fl::event_inside(X, Y, W, H)) {
       deselect();
+    }
```

Comment:
If I were to change anything here, I'd have moved deselect() up to the same line as the event_inside() line, but I think I'd have left the original code if hand editing, just to fix indenting issues.

Reply:
Really? I'm surprised. This change is from your original patch in #121 (https://github.com/fltk/fltk/files/5012528/Fl_Check_Browser--whitespace-patch.txt) - not at all modified by me:
```
   522	 int Fl_Check_Browser::handle(int event) {
   523	   if (event == FL_PUSH) {
   524	     int X, Y, W, H;
   525	     bbox(X, Y, W, H);
   526	-    if (Fl::event_inside(X, Y, W, H))
   527	+    if (Fl::event_inside(X, Y, W, H)) {
   528	       deselect();
   529	+    }
   530	   }
   531	   return Fl_Browser_::handle(event);
```

I agree that the curly braces are not necessary here, but I would still leave this on two lines as it was in the original code.

Reply:
I can't remember -- usually I add braces like that for one liners only when I'm debugging, and I either leave it in place if I think more debugging is necessary, or leave them out when done.

When editing fltk code, I often have an image in my head of Bill's general view of keeping code as tight fitting as possible, hence the curly brace coding style and very conservative use of vertical white separation and two space indent. It's all about getting as much code on the page as possible. In the 90's that was a bigger deal than it is now, but I understand the benefits and try to continue to employ them. My code in Fl_Tree.cxx is an example of that; you'll see little vertical white used in the code and lots of one lines for simple if()s. I only use vertical white in FLTK if I think it helps code comprehension for particularly complicated sections.